conftest.py

```diff
@@ -336,27 +337,53 @@ def data_repository(directory_tree):
     return repo
 
 
+def clone_compressed_repository(base_path, name):
+    """Decompress and clone a repository."""
+
+    compressed_repo_path = Path(__file__).parent / "tests" / "fixtures" / f"{name}.tar.gz"
+    working_dir = base_path / name
+
+    bare_base_path = working_dir / "bare"
+    with tarfile.open(compressed_repo_path, "r") as fixture:
+        fixture.extractall(str(bare_base_path))
+
+    bare_path = bare_base_path / name
+    repository_path = working_dir / "repository"
+    repository = Repo(bare_path, search_parent_directories=True).clone(repository_path)
+
+    return repository
```

Comment:
Should we yield here, and cleanup explicitly after execution?

Reply:
This is used in both session-/function-scoped fixtures so we cannot make it a function-scoped fixture. If we make it session-scoped then we need an extra fixture for the function-scoped fixtures to reset the state after each test. I believe it is cleaner/less code to leave it as a function.

Reply:
Ah, I assumed it might not get cleaned up, but it does - all good! :+1: :see_no_evil: